packages/catalog-process/test/catalogService.integration.test.ts

```diff
@@ -990,6 +983,161 @@ describe("database test", async () => {
         expect(writtenPayload.eServiceDescriptor).toEqual(expectedDescriptorV1);
       });
 
+      it("should also write on event-store to archive the previously published descriptor", async () => {
+        vi.useFakeTimers();
+        vi.setSystemTime(new Date());
+        const descriptor1: Descriptor = {
+          ...mockDescriptor,
+          id: generateId(),
+          state: descriptorState.published,
+          publishedAt: new Date(),
+          interface: mockDocument,
+        };
+        const descriptor2: Descriptor = {
+          ...mockDescriptor,
+          id: generateId(),
+          state: descriptorState.draft,
+          interface: mockDocument,
+        };
+        const eService: EService = {
+          ...mockEService,
+          descriptors: [descriptor1, descriptor2],
+        };
+        await addOneEService(eService, postgresDB, eservices);
+        await catalogService.publishDescriptor(
+          eService.id,
+          descriptor2.id,
+          getMockAuthData(eService.producerId)
+        );
+        const writtenEvents = await readLastTwoEventsByStreamId(
+          eService.id,
+          postgresDB
+        );
+
+        const publicationEvent = writtenEvents[0];
+        const archivingEvent = writtenEvents[1];
+        expect(archivingEvent.stream_id).toBe(eService.id);
+        expect(archivingEvent.version).toBe("1");
+        expect(archivingEvent.type).toBe("EServiceDescriptorUpdated");
+        const archivingPayload = decodeProtobufPayload({
+          messageType: EServiceDescriptorUpdatedV1,
+          payload: archivingEvent.data,
+        });
+        expect(publicationEvent.stream_id).toBe(eService.id);
+        expect(publicationEvent.version).toBe("2");
+        expect(publicationEvent.type).toBe("EServiceDescriptorUpdated");
```

Comment:
```suggestion
        expect(archivingEvent).toMatchObject({
          stream_id: eService.id,
          version: "1",
          type: "EServiceDescriptorUpdated",
        });
```
I think that using `.toMatchObject` is cleaner in these cases, you can do the same for the others in the rest of the tests

Reply:
If you prefer I can edit this test, but I would like to avoid the refactor of the entire suite in this PR.

Reply:
Yes edit only the test touched in this PR, don't refactor the whole suite because it's out of scope of the PR

Reply:
Updated